tests/functional_test_base.go

```diff
@@ -217,7 +219,7 @@ func checkTestShard(t *testing.T) {
 		t.Fatal("Couldn't convert TEST_SHARD_INDEX")
 	}
 
-	// This was determined empirically to distribute our existing test names + run times
+	// This was determined empirically to distribute our existing test names
 	// reasonably well. This can be adjusted from time to time.
 	// For parallelism 4, use 11. For 3, use 26. For 2, use 20.
 	const salt = "-salt-26"
```

Comment:
If you're doing by sub-test you can just remove the salt and use the name directly. It was just there to force the large ones to fall in separate shards

Reply:
I tried removing, but it actually distributed the tests quite unevenly. With salt was better.

Reply:
Huh, weird. I came up with that value by collecting run times and writing a little program to search for the value that made the best distribution. Maybe we should do that again but with run times for the individual tests? It's unlikely the same value is best for suite names vs test names.

(Of course best would be to have it adjust based on historical data.. a previous startup I worked at solved this exact problem. But not worth it here.)